File: subcommands/src/test/java/in/kyle/mcspring/subcommands/TestConsole.java
package in.kyle.mcspring.subcommands;

import org.bukkit.command.CommandSender;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.function.Consumer;

import in.kyle.mcspring.command.SimpleMethodInjection;
import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class TestConsole {
    
    private final SimpleMethodInjection injection;
    
    public void run(CommandSender sender, String commandString, Consumer<PluginCommand> consumer) {
        List<String> parts = new ArrayList<>(Arrays.asList(commandString.split(" ")));
        if (parts.get(0).isEmpty()) {
            parts.remove(0);
        }
        PluginCommand command = new PluginCommand(injection, sender, parts, new ArrayList<>());
        consumer.accept(command);
    }
}

File: mcspring-api/mcspring-subcommands/src/test/java/in/kyle/mcspring/subcommands/TestConsole.kt
package `in`.kyle.mcspring.subcommands

import `in`.kyle.mcspring.command.SimpleMethodInjection
import `in`.kyle.mcspring.subcommands.plugincommand.api.PluginCommand
import `in`.kyle.mcspring.subcommands.plugincommand.impl.PluginCommandImpl
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.mockito.Mockito.mock

object TestConsole {

    fun runCommand(commandString: String,
                   consumer: (PluginCommand) -> Unit,
                   runExecutors: Boolean = true,
                   sender: CommandSender = mock(Player::class.java)): PluginCommandImpl {
        val command = makeCommand(commandString, sender, runExecutors)
        consumer(command)
        return command
    }

    fun makeCommand(command: String,
                            player: CommandSender = mock(Player::class.java),
                            runExecutors: Boolean = true):
            PluginCommandImpl {
        return PluginCommandImpl(SimpleMethodInjection(listOf()),
                player,
                command.split(" ").filter { it.isNotBlank() }.toMutableList(),
                runExecutors)
    }
}
